File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/WriteVariableValueRequest.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.types.common.UaObjectId;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.Map;

public class WriteVariableValueRequest {
    private final UaObjectId objectId;
    private final Map<UaChildId, Variant> variableValues;

    public WriteVariableValueRequest(
            UaObjectId objectId,
            Map<UaChildId, Variant> variableValues)
    {
        this.objectId = objectId;
        this.variableValues = variableValues;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public Map<UaChildId, Variant> getVariableValues() {
        return variableValues;
    }
}

File: opcua-webservice-node/source/src/server/service/message/reactiveobject/WriteVariableValueRequest.ts
import { UaVariant } from "opcua-webapi-ts";
import { UaObjectId } from "../../../types";

export class WriteVariableValueRequest {
    private readonly _objectId: UaObjectId;
    private readonly _variableValues: Map<string, UaVariant>;

    constructor(
        objectId: UaObjectId,
        variableValues: Map<string, UaVariant>,
    ) {
        this._objectId = objectId;
        this._variableValues = variableValues;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get variableValues(): Map<string, UaVariant> {
        return this._variableValues;
    }
}